guacamole/src/main/frontend/src/app/storage/services/localStorageService.js

```diff
@@ -119,10 +122,18 @@ angular.module('storage').provider('localStorageService', [function localStorage
         }
         catch (ignore) {}
 
-        // Pull and parse value from internal storage, if present
+        // Pull from internal storage, if present
         var data = storedItems[key];
-        if (data)
-            return JSON.parse(data);
+        if (data) {
+
+            // Serialize to JSON if possible
+            try {
+                return JSON.stringify(value);
+            } catch (ignore) {}
```

Comment:
I think you mean `JSON.parse()` here, but I also don't think we should attempt to parse values that we expect may not be JSON, as doing so introduces ambiguity (a value intended to be stored as a string may happen to also be valid JSON).

We could universally store everything as JSON, in which case a string would just be stored as a JSON string, or we could provide functions that are specifically intended for storage/retrieval of only string values.

Reply:
> A raw string isn't a valid JSON-parseable value, ...

Isn't it? I get:

![Stringifying and parsing the JavaScript string 'foo'](https://user-images.githubusercontent.com/4632905/204666024-f49e770b-b5f8-4495-9465-a2178a8cd671.png)

Reply:
Err, yeah. I'm not sure what was going on when I tested that earlier.

Reply:
Well, that was a bunch of ado about nothing. Removed my changes to this service.